### PR TITLE
Update and extend CI

### PR DESCRIPTION
**Pull request by Bodigrim**
Adding Windows and macOS builds appeared to be quite a challenge, sigh.

### CODE REVIEW

**Comment by Bodigrim** [comment]
Windows build hangs in `core-tests/resource-release-test.sh`, will investigate tomorrow.

**Comment by Bodigrim** [comment]
This is finally ready for review.
CC @andreasabel @VictorCMiraldo

**Review by andreasabel — APPROVED** [review]
Thanks, great work!
Approved, with the request to add a comment on `MSYS_NO_PATHCONV`.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -54,8 +67,13 @@ jobs:
 
         (cd core-tests && tasty-core-tests +RTS -N2)
         core-tests/exit-status-tests.sh
+        MSYS_NO_PATHCONV=1 core-tests/failing-pattern-test.sh
```
[the remark this answers is not in the document]
Comment (Bodigrim) [reply .github/workflows/ci.yaml:71]: Thanks, done.

**Pull request merged**